### PR TITLE
Iterate zoom out shuffle into a more visual control

### LINKED ISSUE #66105: Iterate zoom out shuffle into a more visual control
Currently, the shuffle control in Gutenberg allows users to iterate through like patterns to pick from. This proposal aims to improve the feature by transforming it into a more visual selecting experience, enhancing usability and providing a more intuitive interface.

### Current 

Image

### Proposed

Replace the existing shuffle control with more visual picker that displays thumbnail previews of patterns in a grid within a popover, instead of iterating to the next pattern. I suggest a max rendering of six, to keep scope minimal here. 

Image

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Resolves: https://github.com/WordPress/gutenberg/issues/66105

This PR will update the `shuffle` feature for patterns with a more visual control that shows the previews of some available patterns from the same pattern category.

I [commented](https://github.com/WordPress/gutenberg/issues/66105#issuecomment-2418699620) the below in the issue: 

> It's worth noting that we have other similar patterns lists like:
>1. Quick inserter previews
>2. `Replace` in blocks like Query Loop (bigger dialog)

> This issue suggests yet another design. Should we proceed with the new design and see if can consolidate designs or this is considered the best one for this specific functionality?

--cc @WordPress/gutenberg-design 

Currently I'm using the `BlockPatternsList` - same with quick inserter.

## Testing Instructions
1. Check that there is no `shuffle` button and in its place we render the `change design` one.
2. Changing patterns should work as expected.

## Screenshots or screencast 
TBD

### CODE REVIEW

**Comment by jasmussen** [comment]
Quick test with a GIF:

test

The new text button is available only when zoomed out, which can be a good way to introduce this without necessarily touching the shuffle button. I'm happy to try this as an early aspect of 6.8.

That said, outside of any other dropdown label suggestions, it should be "Change design", sentencecase rather than titlecase.

**Comment by ntsekouras** [comment]
>The new text button is available only when zoomed out, which can be a good way to introduce this without necessarily touching the shuffle button.

Currently this PR removes the `shuffle` button. Do you meant to preserve both?

**Comment by youknowriad** [comment]
Should we take this as an opportunity to unify all these "switch/change/replace" flows? For clarity I'm not asking for this to be done as part of this PR necessarily, just wondering whether it makes sense and if it does, if we should do it (in steps)

**Comment by ntsekouras** [comment]
>Should we take this as an opportunity to unify all these "switch/change/replace" flows? For clarity I'm not asking for this to be done as part of this PR necessarily, just wondering whether it makes sense and if it does, if we should do it (in steps)

I 100% agree!

**Comment by jameskoster** [comment]
> Should we take this as an opportunity to unify all these "switch/change/replace" flows?

Possibly related to this: https://github.com/WordPress/gutenberg/issues/40208#issuecomment-2383010771

**Comment by richtabor** [comment]
> The new text button is available only when zoomed out, which can be a good way to introduce this without necessarily touching the shuffle button. I'm happy to try this as an early aspect of 6.8.

Shuffle is only available in zoom out.

**Comment by richtabor** [comment]
> Currently this PR removes the shuffle button. Do you meant to preserve both?

No, this "Change Design" button is essentially a second iteration of the shuffle button.

**Comment by richtabor** [comment]
> Should we take this as an opportunity to unify all these "switch/change/replace" flows? For clarity I'm not asking for this to be done as part of this PR necessarily, just wondering whether it makes sense and if it does, if we should do it (in steps)

Yes. `BlockPatternsList` could be leveled-up as well to not be so heavy (masonry style layout option, no title option).

**Comment by youknowriad** [comment]
Yep, I think we should consider creating a storybook story for the `BlockPatternsList` component to be able to work on it in isolation.

**Comment by ntsekouras** [comment]
So I guess we agree on using `BlockPatternsList` component now and iterate on that.

>Yep, I think we should consider creating a storybook story for the BlockPatternsList component to be able to work on it in isolation.

I can create a PR for that.

**Comment by ntsekouras** [comment]
So what's missing from this PR? Do we want to update some designs or we'll do it follow ups, consolidating the different similar lists?

**Comment by ellatrix** [comment]
It would be cool if the popover was kept open, and somehow don't let it overlap with the pattern shown, so you can click around more easily.

**Comment by richtabor** [comment]
> It would be cool if the popover was kept open, and somehow don't let it overlap with the pattern shown, so you can click around more easily.

Perhaps if it stayed open, it could render above if there's room. A follow-up to explore?

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 1]
```diff
@@ -285,3 +285,19 @@
 		}
 	}
 }
+
+.block-editor-block-toolbar-change-design-content-wrapper {
+	padding: $grid-unit-15;
+	width: 320px;
+	.block-editor-block-patterns-list {
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-toolbar/style.scss:292]: What's your current thoughts about absorbing this into the component with `variant="grid"` or something. How feasible...?

**Review by youknowriad — APPROVED** [review]
Assume this direction is the right one design wise. This looks good tome. I'd appreciate a follow-up improving the pattern list component.

File: packages/block-editor/src/components/block-toolbar/style.scss (hunk 1 above)
Comment (ntsekouras) [reply packages/block-editor/src/components/block-toolbar/style.scss:292]: I think it's doable (regarding the current usages and wrappers etc..). It's on my list to do this and see to consolidate the similar replace lists.

**Pull request merged**